### DISCUSSION TITLE
Is there a way to deregister a GJS subclass type?

### DISCUSSION THREAD

**Discussion by klangman** [linuxmint/discussions#570] [Development] [answered]
I have some Javascript for a new Cinnamon extension I am writing that use `GObject.registerClass(this)` to register a new subclass type, but I can't seem to find any mention of an API to unregister/deregister the class when it's no longer needed because the user disables the extension. Anyone know if that's possible? Currently I get an exception when the extension is re-enabled due to an attempt to register a class using a previously registered name.

**Comment by rcalixte** [comment, marked as answer]
I dealt with this for another applet. The [source file](https://github.com/linuxmint/cinnamon-spices-applets/blob/2f09496/color-blind-filters%40rcalixte/files/color-blind-filters%40rcalixte/5.4/Shaders.js) should help you here. I ended up just hacking around the class name with the random value and that resolved the class register/deregister conundrum. (You can add logging to the applet and see the class names for how it registers/deregisters if you're curious.)

It's a hack because as you can imagine, it has the potential to waste memory for each register/deregister until Cinnamon is reloaded but as it isn't much memory and the number of events for it to matter would need to be rather extreme, it was a tradeoff I could live with. If your next applet (can't wait!) will require frequent and constant adjusting, it may not work. If the applet's configuration will stabilize, this should hopefully help.

**Comment by klangman** [comment]
I had considered randomizing the name but I was hoping for something less hacky. I am satisfied knowing that I am not alone here and that I am not missing an obvious non-hacky solution that just escaped my google search. Thanks!!
